ui/src/app/base/components/AuthenticationFields/AuthenticationFields.test.tsx

```diff
@@ -0,0 +1,67 @@
+import { mount } from "enzyme";
+import { Formik } from "formik";
+
+import AuthenticationFields from "./AuthenticationFields";
+
+describe("AuthenticationFields", () => {
+  it("renders different fields depending on whether a certificate should be generated", () => {
+    const Proxy = ({ shouldGenerateCert }: { shouldGenerateCert: boolean }) => (
+      <Formik
+        initialValues={{ certificate: "", key: "", password: "" }}
+        onSubmit={jest.fn()}
+      >
+        <AuthenticationFields
+          onRadioChange={jest.fn()}
+          shouldGenerateCert={shouldGenerateCert}
+        />
+      </Formik>
+    );
+    const wrapper = mount(<Proxy shouldGenerateCert={true} />);
+    expect(wrapper.find("input[name='password']").exists()).toBe(true);
+    expect(wrapper.find("textarea[name='certificate']").exists()).toBe(false);
+    expect(wrapper.find("textarea[name='key']").exists()).toBe(false);
+
+    wrapper.setProps({ shouldGenerateCert: false });
+    wrapper.update();
+    expect(wrapper.find("input[name='password']").exists()).toBe(false);
+    expect(wrapper.find("textarea[name='certificate']").exists()).toBe(true);
+    expect(wrapper.find("textarea[name='key']").exists()).toBe(true);
+  });
+
+  it("can be chosen to not render the password field", () => {
+    const Proxy = ({ showPassword }: { showPassword: boolean }) => (
+      <Formik
+        initialValues={{ certificate: "", key: "", password: "" }}
+        onSubmit={jest.fn()}
+      >
+        <AuthenticationFields
+          onRadioChange={jest.fn()}
+          shouldGenerateCert
+          showPassword={showPassword}
+        />
+      </Formik>
+    );
+    const wrapper = mount(<Proxy showPassword={true} />);
+    expect(wrapper.find("input[name='password']").exists()).toBe(true);
+
+    wrapper.setProps({ showPassword: false });
```

Comment:
Again, because we're not testing props changing I think this should be two tests.